src/libvfcfuncinstrument/libVFCFuncInstrument.cpp

```diff
@@ -396,58 +546,60 @@ struct VfclibFunc : public ModulePass {
             haveFloatingPointArithmetic(pi, f, is_from_library, is_intrinsic,
                                         &use_float, &use_double, M);
 
-            if (!(is_intrinsic && (!use_float && !use_double))) {
-              // Create function ID
-              Value *FunctionID = Builder.CreateGlobalStringPtr(FunctionName);
-
-              // Constants creation
-              Constant *isLibraryFunction = ConstantInt::get(
-                  Type::getInt8Ty(M.getContext()), is_from_library);
-              Constant *isInstrinsicFunction = ConstantInt::get(
-                  Type::getInt8Ty(M.getContext()), is_intrinsic);
-              Constant *haveFloat =
-                  ConstantInt::get(Type::getInt8Ty(M.getContext()), use_float);
-              Constant *haveDouble =
-                  ConstantInt::get(Type::getInt8Ty(M.getContext()), use_double);
-
-              // Enter function arguments
-              std::vector<Value *> MetaData{FunctionID, isLibraryFunction,
-                                            isInstrinsicFunction, haveFloat,
-                                            haveDouble};
-
-              Type *ReturnTy = f->getReturnType();
-              std::vector<Type *> CallTypes;
-              for (auto it = pi->op_begin(); it < pi->op_end() - 1; it++) {
-                CallTypes.push_back(cast<Value>(it)->getType());
-              }
-
-              FunctionType *HookFunTy =
-                  FunctionType::get(ReturnTy, CallTypes, false);
-              Function *hook_func = Function::Create(
-                  HookFunTy, Function::ExternalLinkage, NewName, &M);
-
-              hook_func->setAttributes(f->getAttributes());
-              hook_func->setCallingConv(f->getCallingConv());
-
-              BasicBlock *block =
-                  BasicBlock::Create(M.getContext(), "block", hook_func);
-
-              InstrumentFunction(MetaData, hook_func, f, cast<CallInst>(pi),
-                                 block);
-
-              cast<CallInst>(pi)->setCalledFunction(hook_func);
+            if (is_intrinsic && !(use_float || use_double)) {
+              continue;
+            }
+
```

Comment:
Could you detail in the comments a bit more what is happening here ? It's not easy to follow the code.

Reply:
That's because some intrinsic functions can be a problem for instrumentation, so I only instruct intrinsic functions that use floats or doubles. It would be better to make a list of intrinsic functions in LLVM that do floating-point calculations in the future.